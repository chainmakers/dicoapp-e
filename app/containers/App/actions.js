// @flow
/*
 * App Actions
 *
 * Actions change things in your application
 * Since this boilerplate uses a uni-directional data flow, specifically redux,
 * we have these actions which are the only way your application interacts with
 * your application state. This guarantees that your state is up to date and nobody
 * messes it up weirdly somewhere.
 *
 * To add a new Action:
 * 1) Import your constant
 * 2) Add a function like this:
 *    export function yourAction(var) {
 *        return { type: YOUR_ACTION_CONSTANT, var: var }
 *    }
 * https://github.com/redux-utilities/flux-standard-action
 */

import {
  LOGIN,
  LOGIN_SUCCESS,
  LOGIN_ERROR,
  LOGOUT,
  LOAD_BALANCE,
  LOAD_BALANCE_SUCCESS,
  LOAD_BALANCE_ERROR,
  LOAD_COIN_BALANCE_SUCCESS,
  LOAD_WITHDRAW,
  LOAD_WITHDRAW_SUCCESS,
  LOAD_WITHDRAW_ERROR,
  LOAD_SWAP_SUCCESS
} from './constants';
import type { BalancePayload } from './schema';

export function login(passphrase: string) {
  return {
    type: LOGIN,
    payload: {
      passphrase
    }
  };
}

export function loginSuccess(user) {
  return {
    type: LOGIN_SUCCESS,
    payload: {
      user
    }
  };
}

export function loginError(error) {
  return {
    type: LOGIN_ERROR,
    error
  };
}

export function logout() {
  return {
    type: LOGOUT
  };
}

export function loadBalance() {
  return {
    type: LOAD_BALANCE
  };
}

export function loadBalanceSuccess() {
  return {
    type: LOAD_BALANCE_SUCCESS
  };
}

export function loadCoinBalanceSuccess(payload: BalancePayload) {
  return {
    type: LOAD_COIN_BALANCE_SUCCESS,
    payload
  };
}

export function loadBalanceError(message: string) {
  return {
    type: LOAD_BALANCE_ERROR,
    error: {
      message
    }
  };
}

export function loadWithdraw(payload: {
  amount: number,
  address: string,
  coin: string
}) {
  return {
    type: LOAD_WITHDRAW,
    payload
  };
}

export function loadWithdrawSuccess(payload: {
  amount: number,
  address: string,
  coin: string
}) {
  return {
    type: LOAD_WITHDRAW_SUCCESS,
    payload
  };
}

export function loadWithdrawError(
  payload: {
    amount: number,
    address: string,
    coin: string
  },
  message: string
) {
  return {
    type: LOAD_WITHDRAW_ERROR,
    payload,
    error: {
      message
    }
  };
}

// eslint-disable-next-line flowtype/no-weak-types
export function loadSwapSuccess(payload: Object) {
  return {
    type: LOAD_SWAP_SUCCESS,
    payload
  };
}

// -- //
