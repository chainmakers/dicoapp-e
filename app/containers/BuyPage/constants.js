// eslint-disable-next-line import/prefer-default-export
export const APP_STATE_NAME = 'buy';
export const LOAD_PRICES = 'dicoapp/BuyPage/LOAD_PRICES';
export const LOAD_PRICES_SUCCESS = 'dicoapp/BuyPage/LOAD_PRICES_SUCCESS';
export const LOAD_PRICES_ERROR = 'dicoapp/BuyPage/LOAD_PRICES_ERROR';
export const LOAD_PRICE = 'dicoapp/BuyPage/LOAD_PRICE';
export const LOAD_BEST_PRICE = 'dicoapp/BuyPage/LOAD_BEST_PRICE';
export const LOAD_BUY_COIN = 'dicoapp/BuyPage/LOAD_BUY_COIN';
export const LOAD_BUY_COIN_SUCCESS = 'dicoapp/BuyPage/LOAD_BUY_COIN_SUCCESS';
export const LOAD_BUY_COIN_ERROR = 'dicoapp/BuyPage/LOAD_BUY_COIN_ERROR';
export const CLEAR_BUY_COIN_ERROR = 'dicoapp/BuyPage/CLEAR_BUY_COIN_ERROR';
export const LOAD_RECENT_SWAPS = 'dicoapp/BuyPage/LOAD_RECENT_SWAPS';
export const LOAD_RECENT_SWAPS_COIN = 'dicoapp/BuyPage/LOAD_RECENT_SWAPS_COIN';
export const LOAD_RECENT_SWAPS_SUCCESS =
  'dicoapp/BuyPage/LOAD_RECENT_SWAPS_SUCCESS';
export const LOAD_RECENT_SWAPS_DATA_FROM_WEBSOCKET =
  'dicoapp/BuyPage/LOAD_SWAP_DATA_FROM_WEBSOCKET';
export const LOAD_RECENT_SWAPS_ERROR =
  'dicoapp/BuyPage/LOAD_RECENT_SWAPS_ERROR';
export const CHECK_TIMEOUT_EVENT = 'dicoapp/BuyPage/CHECK_TIMEOUT_EVENT';
export const CHECK_UPDATE_SWAP_EVENT =
  'dicoapp/BuyPage/CHECK_UPDATE_SWAP_EVENT';
export const AUTO_HIDE_SNACKBAR_TIME = 6000;
export const TIME_LOOP = 20 * 1000;
export const STATE_SWAPS = [
  'Confirming',
  'My Fee',
  'Bob Deposit',
  'Alice Payment',
  'Bob Payment',
  'Swap Finished' // 'Alice Spend'
];

// https://github.com/KomodoPlatform/Documentation/blob/master/docs/source/barterDEX/MarketmakerErrors.rst#marketmaker-error-codes
export const APPROPRIATE_ERROR_UTXOS =
  'cant find a deposit that is close enough in size. make another deposit that is just a bit larger than what you want to trade';
export const SWAP_TX_DEFAULT =
  '0000000000000000000000000000000000000000000000000000000000000000';

export const SWAP_TIMEOUT = 'dicoapp/BuyPage/SWAP_TIMEOUT';
export const SWAP_MAKE_A_NEW = 'dicoapp/BuyPage/SWAP_MAKE_A_NEW';
export const SWAP_DETAIL_MODAL_OPEN = 'dicoapp/BuyPage/SWAP_DETAIL_MODAL_OPEN';
export const SWAP_DETAIL_MODAL_CLOSE =
  'dicoapp/BuyPage/SWAP_DETAIL_MODAL_CLOSE';
