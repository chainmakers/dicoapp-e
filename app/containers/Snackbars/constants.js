// eslint-disable-next-line import/prefer-default-export
export const APP_STATE_NAME = 'snackbars';

export const SNACKBARS_OPEN = 'dicoapp/BuyPage/SNACKBARS_OPEN';
export const SNACKBARS_CLOSE = 'dicoapp/BuyPage/SNACKBARS_CLOSE';
