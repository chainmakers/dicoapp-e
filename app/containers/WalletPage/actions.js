// @flow

import {
  LOAD_TRANSACTIONS_SUCCESS,
  LOAD_TRANSACTIONS_ERROR,
  WITHDRAW_MODAL_OPEN,
  WITHDRAW_MODAL_CLOSE,
  DEPOSIT_MODAL_OPEN,
  DEPOSIT_MODAL_CLOSE,
  TRANSACTIONS_LOAD,
  TRANSACTIONS_LOAD_LOOP,
  COIN_TRANSACTIONS_LOAD,
  COIN_TRANSACTIONS_SUCCESS
} from './constants';
import type {
  CoinTransactionsLoadPayload,
  CoinTransactionsSuccessPayload
} from './schema';

export function loadTransactionsSuccess() {
  return {
    type: LOAD_TRANSACTIONS_SUCCESS
  };
}

export function loadTransactionsError(message: string) {
  return {
    type: LOAD_TRANSACTIONS_ERROR,
    error: {
      message
    }
  };
}

export function openWithdrawModal(coin: string) {
  return {
    type: WITHDRAW_MODAL_OPEN,
    payload: {
      coin
    }
  };
}

export function closeWithdrawModal() {
  return {
    type: WITHDRAW_MODAL_CLOSE
  };
}

export function openDepositModal(coin: string) {
  return {
    type: DEPOSIT_MODAL_OPEN,
    payload: {
      coin
    }
  };
}

export function closeDepositModal() {
  return {
    type: DEPOSIT_MODAL_CLOSE
  };
}

export function loadTransactions() {
  return {
    type: TRANSACTIONS_LOAD
  };
}

export function loadTransactionsLoop() {
  return {
    type: TRANSACTIONS_LOAD_LOOP
  };
}

export function loadCoinTransactions(payload: CoinTransactionsLoadPayload) {
  return {
    type: COIN_TRANSACTIONS_LOAD,
    payload
  };
}

export function loadCoinTransactionsSuccess(
  payload: CoinTransactionsSuccessPayload
) {
  return {
    type: COIN_TRANSACTIONS_SUCCESS,
    payload
  };
}
