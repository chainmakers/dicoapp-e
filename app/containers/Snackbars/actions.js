// @flow

import { SNACKBARS_OPEN, SNACKBARS_CLOSE } from './constants';

export function openSnackbars(message: string) {
  return {
    type: SNACKBARS_OPEN,
    payload: {
      message
    }
  };
}

export function closeSnackbars() {
  return {
    type: SNACKBARS_CLOSE
  };
}
