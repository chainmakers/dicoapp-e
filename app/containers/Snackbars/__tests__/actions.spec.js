import { openSnackbars, closeSnackbars } from '../actions';

import { SNACKBARS_OPEN, SNACKBARS_CLOSE } from '../constants';

describe('containers/BuyPage/actions/openSnackbars', () => {
  const message = 'message';
  it('should openSnackbars should create openSnackbars action', () => {
    expect(openSnackbars(message)).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: SNACKBARS_OPEN,
      payload: {
        message
      }
    };

    expect(openSnackbars(message)).toEqual(expectedResult);
  });
});

describe('containers/BuyPage/actions/closeSnackbars', () => {
  it('should closeSnackbars should create closeSnackbars action', () => {
    expect(closeSnackbars()).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: SNACKBARS_CLOSE
    };

    expect(closeSnackbars()).toEqual(expectedResult);
  });
});
