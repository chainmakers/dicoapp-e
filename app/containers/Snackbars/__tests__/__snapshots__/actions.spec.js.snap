// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/BuyPage/actions/closeSnackbars should closeSnackbars should create closeSnackbars action 1`] = `
Object {
  "type": "dicoapp/BuyPage/SNACKBARS_CLOSE",
}
`;

exports[`containers/BuyPage/actions/openSnackbars should openSnackbars should create openSnackbars action 1`] = `
Object {
  "payload": Object {
    "message": "message",
  },
  "type": "dicoapp/BuyPage/SNACKBARS_OPEN",
}
`;
